### ISSUE TITLE
Support adding guide line for zero value

### ISSUE THREAD

**Issue by kanitw** [vega/vega-lite#373]
![tableau_-_book2](https://cloud.githubusercontent.com/assets/111269/6631597/f34e4d86-c8e5-11e4-8bf8-fe186530b9cd.png)
- [ ] Need to add another mark group with static value

**Comment by willium** [comment]
Going to get started on this tomorrow, but I had a couple clarifying questions:
Is there a good dataset to see this and test on?
Is there another, already implemented, mark group that I should model this on?
Thanks!

**Comment by kanitw** [comment]
@willium  upon thinking more, I think part of this should be handled at the Vega level, so this might not be the best starter task.  I just assigned another issue about “labeling" to you.  That might be more fun too.

**Comment by kanitw** [comment]
One way to achieve this is to support #594.

**Comment by kanitw** [comment]
This would require `datum` instead of `value` though and thus is blocked by #1601

**Comment by Fil** [comment]
What can we do to add this feature?

**Comment by domoritz** [comment]
This is possible with layering and well supported in 2.0. 

```
{
  "$schema": "https://vega.github.io/schema/vega-lite/v2.json",
  "data": {"url": "data/cars.json"},
  "layer": [
    {
      "mark": "point",
      "encoding": {
        "x": {"field": "Horsepower","type": "quantitative"},
        "y": {"field": "Miles_per_Gallon","type": "quantitative"}
      }
    },
    {
      "data": {"values": [{"guide": 20}]},
      "mark": "rule",
      "encoding": {
        "y": {"field": "guide","type": "ordinal"},
        "color": {"value": "red"}
      }
    }
  ]
}
```

I wonder whether we want to automatically add lines for 0. 

<img width="252" alt="screen shot 2017-04-19 at 12 47 54" src="https://cloud.githubusercontent.com/assets/589034/25199086/75fa4598-24fe-11e7-92e5-ad9af4f4fa83.png">

**Comment by Fil** [comment]
Sorry I realise this is vega-light; I've managed to create what I wanted with the following under vega:
```
  "marks": [
    {
      "type": "rule",
      "encode": {
        "enter": {
          "x": {"value": 0},
          "x2": {"signal": "width"},
          "y": {"scale": "y", "value": 0},
          "stroke": {"value": "#000000"},
          "strokeWidth": {"value": 2},
        }
      }
    }
  …
  ]
```

**Comment by domoritz** [comment]
Excellent. But Vega-Lite can definitely support the use case in this issue.

**Comment by Fil** [comment]
btw I couldn't find the `"guide"` keyword in the code or in the documentation.
Is there a complete list of keywords somewhere?

**Comment by domoritz** [comment]
@Fil We will release Vega-Lite 2.0 next week. The `rule` mark will be added to the mark docs. `guide` is not a keyword. 

The schema (which is generated from the code) has a list of all primitive marks: https://github.com/vega/schema/blob/master/vega-lite/v2.0.0-alpha.9.json#L2429. Note that Vega-Lite 2.0 also support composite marks such as boxplots.

**Comment by domoritz** [comment]
This works in Vega-Lite 2.

**Comment by domoritz** [comment]
I think we should just add an id scale so this becomes easier.

**Comment by kanitw** [comment]
We already have an issue for `datum` at https://github.com/vega/vega-lite/issues/1601.

 https://github.com/vega/vega-lite/pull/3239 also provides another way to do this by using condition with grid encoding.
